src/hotspot/share/memory/arena.cpp

```diff
@@ -129,28 +129,31 @@ Chunk* ChunkPool::allocate_chunk(size_t length, AllocFailType alloc_failmode) {
          SIZE_FORMAT ".", length);
   // Try to reuse a freed chunk from the pool
   ChunkPool* pool = ChunkPool::get_pool_for_size(length);
+  Chunk* chunk = nullptr;
   if (pool != nullptr) {
     Chunk* c = pool->take_from_pool();
     if (c != nullptr) {
       assert(c->length() == length, "wrong length?");
-      return c;
+      chunk = c;
     }
   }
-  // Either the pool was empty, or this is a non-standard length. Allocate a new Chunk from C-heap.
-  size_t bytes = ARENA_ALIGN(sizeof(Chunk)) + length;
-  void* p = os::malloc(bytes, mtChunk, CALLER_PC);
-  if (p == nullptr && alloc_failmode == AllocFailStrategy::EXIT_OOM) {
-    vm_exit_out_of_memory(bytes, OOM_MALLOC_ERROR, "Chunk::new");
+  if (chunk == nullptr) {
+    // Either the pool was empty, or this is a non-standard length. Allocate a new Chunk from C-heap.
+    size_t bytes = ARENA_ALIGN(sizeof(Chunk)) + length;
+    void* p = os::malloc(bytes, mtChunk, CALLER_PC);
+    if (p == nullptr && alloc_failmode == AllocFailStrategy::EXIT_OOM) {
+      vm_exit_out_of_memory(bytes, OOM_MALLOC_ERROR, "Chunk::new");
+    }
+    chunk = (Chunk*)p;
   }
-  Chunk* chunk = ::new(p) Chunk(length);
+  ::new(chunk) Chunk(length);
```

Comment:
You need to do this because if you find and take a chunk out of the pool, it's not properly initialized?

Reply:
Okay, I read the code before the comments in the PR, so that answers my question.